src/app/settings/components/project-role-policies-edit/project-role-policies-edit.tsx

```diff
@@ -0,0 +1,181 @@
+import * as React from 'react';
+import * as ReactForm from 'react-form';
+
+import { DataLoader } from '../../../shared/components';
+import { Application } from '../../../shared/models';
+import { services } from '../../../shared/services';
+
+interface ProjectRolePoliciesProps {
+    projName: string;
+    roleName: string;
+    policies: string[];
+    formApi: ReactForm.FormApi;
+    newRole: boolean;
+}
+
+function generatePolicy(project: string, role: string, action?: string, object?: string, permission?: string): string {
+    if (action === undefined) {
```

Comment:
You could use `action || ''` instead of `if (action === undefined)`. I think it is easier to read:

`p,proj:${project}:${role},applications,${action || ''},${project}/${object || ''},${permission || ''}`

Reply:
That looks way better thanks!